lib/common/xml.c

```diff
@@ -915,25 +912,32 @@ xml_log_patchset(uint8_t log_level, const char *function, xmlNode * patchset)
                     } else if(strcmp(op, "set") == 0) {
                         const char *value = crm_element_value(child, "value");
 
-                        if(o_set > 0) {
-                            o_set += snprintf(buffer_set + o_set, XML_BUFFER_SIZE - o_set, ", ");
+                        if (buffer_set->len) {
+                            g_string_append(buffer_set, ", ");
                         }
-                        o_set += snprintf(buffer_set + o_set, XML_BUFFER_SIZE - o_set, "@%s=%s", name, value);
+                        g_string_append_printf(buffer_set, "@%s=%s", name, value);
 
                     } else if(strcmp(op, "unset") == 0) {
-                        if(o_unset > 0) {
-                            o_unset += snprintf(buffer_unset + o_unset, XML_BUFFER_SIZE - o_unset, ", ");
+                        if (buffer_unset->len) {
+                            g_string_append(buffer_unset, ", ");
                         }
-                        o_unset += snprintf(buffer_unset + o_unset, XML_BUFFER_SIZE - o_unset, "@%s", name);
+                        g_string_append_printf(buffer_unset, "@%s", name);
                     }
                 }
-                if(o_set) {
-                    do_crm_log_alias(log_level, __FILE__, function, __LINE__, "+  %s:  %s", xpath, buffer_set);
+                if (buffer_set->len) {
+                    do_crm_log_alias(log_level, __FILE__, function, __LINE__,
+                                     "+  %s:  %s", xpath,
+                                     (const char *) buffer_set->str);
                 }
-                if(o_unset) {
-                    do_crm_log_alias(log_level, __FILE__, function, __LINE__, "-- %s:  %s", xpath, buffer_unset);
+                if (buffer_unset->len) {
+                    do_crm_log_alias(log_level, __FILE__, function, __LINE__,
+                                     "-- %s:  %s", xpath,
+                                     (const char *) buffer_unset->str);
```

Comment:
buffer is only used for logging here too, but the function already returns early if the specified log level isn't active, so there's no problem